specification/sbi_atee.adoc

```diff
@@ -101,35 +101,106 @@ This operation requires the host to convert a sufficient number of non-confident
+[source, C]
+-------------------
+struct nacl_shmem {
+    /* Scratch space. The layout of this scratch space is defined by the particular function being
+     * invoked.
+     *
+     * For the `sbi_tee_host_run_tvm_vcpu()` function in the TEE-H extension, the layout of this 
+     * scratch space matches the `tsm_shmem_scratch` struct given below.
+     */
+    uint64_t scratch[256];
+    uint64_t _reserved[240];
+    /* Bitmap indicating which CSRs in `csrs` the host wishes to sync.
+     *
+     * Currently unused in the TEE-related extensions and will not be read or written by the TSM.
+     */
+    uint64_t dirty_bitmap[16];
+    /* Hypervisor and virtual-supervisor CSRs. The 12-bit CSR number is transformed into a 10-bit
+     * index by extracting bits `{csr[11:10], csr[8:0]}` since `csr[9:8]` is always 2'b10 for HS
+     * and VS CSRs.
+     *
+     * These CSRs may be updated by `sbi_tee_host_run_tvm_vcpu()` in the TEE-Host extension. See
+     * the documentation of `sbi_tee_host_run_tvm_vcpu()` for more details.
+     */
+    uint64_t csrs[1024];
+};
+
+struct tsm_shmem_scratch {
+    /* General purpose registers for a TVM guest.
+     *
+     * The TSM will always read or write the minimum number of registers in this set to complete
+     * the requested action. To avoid leaking information from the TVM, the TSM must follow the
+     * given rules.
+     *
+     * The TSM will write to these registers upon return from `sbi_tee_host_run_tvm_vcpu()` when:
+     *  - The vCPU takes a store guest page fault in an emulated MMIO region.
+     *  - The vCPU makes an ECALL that is to be forwarded to the host.
+     *
+     * The TSM will read from these registers when:
+     *  - The vCPU takes a load guest page fault in an emulated MMIO region.
+     */
+    uint64_t guest_gprs[32];
+    uint64_t _reserved[224];
+};
+-------------------
+
+Given is the list of CSRs and GPRs that TSM and host are intended to use from NACL shared memory. Also given
+are the allowed operation for each software in terms of `R` (read) and `W` (write). For example, the host is only
+supposed to read `htinst`. The host can write to it without any problems but TSM is not supposed to read that
+value as it's only supposed to write to it and ignore any modifications.
+
+[#table_tsm_csr_updates_in_nacl]
+.TSM NACL CSRs and GPRs
+[cols="2,1,1,12", width=100%, align="center", options="header"]
+|===
+| *CSRs*     | *TSM* | *Host* | *Purpose*
+| htinst     |  W    |  R     | In case of MMIO emulation, TSM writes a modified form of load/store
+                                instruction in htinst CSR to allow the host to emulate the MMIO.
+| htval      |  W    |  R     | In case of a guest page-fault, TSM writes the guest physical address into
+                                htval CSR.
+| htimedelta |  W    |  R     | TSM writes the guest htimedelta relative to host's htimedelta in this CSR.
```

Comment:
The host having an `htimedelta` value is an implementation detail of Salus since the host itself runs in VS. I'd remove mention of it here.